src/ts/grid-template.ts

```diff
@@ -0,0 +1,150 @@
+export class GridTemplate {
+
+    constructor(private columnNames: string[], private dataRecords: string[][]) {
+        $(".myGrid").css({
+            "display": "block",
+            "width": "100%",
+            "height": `${100 * 5 / 6}%`,
+            "text-align": "center"
+        });
+
+        $(".controls").css({
+            "display": "block",
+            "width": "100%",
+            "height": `${100 / 6}%`,
+            "position": "fixed",
+            "bottom": "0px"
+        });
+
+        const lable = document.createElement("lable") as HTMLLabelElement;
+        lable.htmlFor = "#search";
+        lable.innerText = "Search: ";
+        const input = document.createElement("input");
```

Comment:
I think you can just do most of these tags in the HTML and use document.getElementByID() for edits rather than crafting the items here, it will also make it easier to see the layout